src/main/java/io/logz/log4j2/LogzioAppender.java

```diff
@@ -216,8 +216,8 @@ private LogzioAppender(String name, Filter filter, final boolean ignoreException
                              String bufferDir, int socketTimeout, int connectTimeout, boolean addHostname,
                              String additionalFields, boolean debug, int gcPersistedQueueFilesIntervalSeconds) {
         super(name, filter, null, ignoreExceptions);
-        this.logzioToken = token;
-        this.logzioUrl = url;
+        this.logzioToken = getValueFromSystemEnvironmentIfNeeded(token);
+        this.logzioUrl = getValueFromSystemEnvironmentIfNeeded(url);
```

Comment:
The function getValueFromSystemEnvironmentIfNeeded() should be called only one time for each property. Therefore it should be deleted from  setLogzioUrl() function (133-135 lines)